src/operator/nn/softmax-inl.h

```diff
@@ -113,6 +113,60 @@ inline void Softmax(Stream<cpu> *s, DType *in, OType *out,
   }
 }
 
+template<typename OP, bool negate, typename AType, typename DType, typename OType,
+         typename IType, int ndim>
+inline void SoftmaxWithLength(Stream<cpu> *s, DType *in, OType *out, IType *length,
```

Comment:
Can we merge this function with Softmax (treat length = M in the default case). Creating two functions just to serve one extra argument seems to be an overkill

Reply:
Answered below.